FreeRTOS_ARP.c

```diff
@@ -135,83 +135,93 @@ eFrameProcessingResult_t eARPProcessPacket( NetworkBufferDescriptor_t * pxNetwor
         NetworkEndPoint_t * pxSourceEndPoint;
     #endif
 
-    /* The field ulSenderProtocolAddress is badly aligned, copy byte-by-byte. */
+    /* Only Ethernet hardware type is supported.
+     * Only IPv4 address can be present in the ARP packet.
+     * The hardware length (the MAC address) must be 6 bytes. And,
+     * The Protocol address length must be 4 bytes as it is IPv4. */
+    if( ( pxARPHeader->usHardwareType == ipARP_HARDWARE_TYPE_ETHERNET ) &&
+        ( pxARPHeader->usProtocolType == ipARP_PROTOCOL_TYPE ) &&
+        ( pxARPHeader->ucHardwareAddressLength == ipMAC_ADDRESS_LENGTH_BYTES ) &&
+        ( pxARPHeader->ucProtocolAddressLength == ipIP_ADDRESS_LENGTH_BYTES ) )
+    {
```

Comment:
I wrote:
~~~c
else
{
    iptraceDROPPED_INVALID_PACKET( pxARPHeader );
}
~~~
which should be `iptraceDROPPED_INVALID_ARP_PACKET()`.

Reply:
Done, thank you.